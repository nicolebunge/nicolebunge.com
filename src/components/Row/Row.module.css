.row {
  grid-auto-flow: dense;
  grid-gap: 3%;
  grid-template-columns: repeat(12, 1fr);
}

@media (width >= 640px) {
  .row {
    display: grid;
  }
}
