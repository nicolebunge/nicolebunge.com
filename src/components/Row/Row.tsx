import clsx from 'clsx';
import React, { ReactNode } from 'react';
import Alignment, { AlignmentProps } from '../Alignment';
import { row } from './Row.module.css';

interface RowProps extends AlignmentProps {
  children: ReactNode;
}

function Row(props: RowProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <Alignment className={clsx(className, row)} {...otherProps} />;
}

export type { RowProps };
export default Row;
