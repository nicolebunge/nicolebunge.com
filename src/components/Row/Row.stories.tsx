import Row from '.';

const meta = {
  component: Row,
};

const Default = {
  args: {
    children: 'Row',
  },
};

export { Default };
export default meta;
