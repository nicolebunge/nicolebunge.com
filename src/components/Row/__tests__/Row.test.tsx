import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import React from 'react';
import { describe, expect, it } from 'vitest';
import * as stories from '../Row.stories';

const { Default } = composeStories(stories);

describe('Row component', () => {
  it('renders correctly', async () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
