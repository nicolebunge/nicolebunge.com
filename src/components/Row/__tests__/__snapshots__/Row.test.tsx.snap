// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Row component > renders correctly 1`] = `
<div
  class="row"
>
  Row
</div>
`;
